bool shouldPlan
string refPos
string actPos
bool executing
bool planning
